src/pages/Profile.js

```diff
@@ -147,7 +154,12 @@ export default function Profile(props) {
       subtitle: result.content.ext.subtitle,
       ts: result.ts,
     };
-    togglePinPost(resolverJSON, result.content.ext.postPath, mySky);
+    const res = await togglePinPost(
+      resolverJSON,
+      result.content.ext.postPath,
+      mySky
+    );
+    setPinStatus(res.success ? "success" : "error");
```

Comment:
I think this is one place that needs to be updated. If there is an error it should make sure the post is not showing in the pinned section.